src/main/java/org/walkmod/javalang/compiler/types/ClassPathTree.java

```diff
@@ -0,0 +1,109 @@
+package org.walkmod.javalang.compiler.types;
+
+
+import sun.misc.URLClassPath;
+
+import java.io.File;
+import java.util.*;
+
+public class ClassPathTree {
```

Comment:
The add method looks strange to me. I suggest writing a test that shows application of the class. Especially dealing with different depths ok key pathes. 
I would call the keys parameter a keyPath.
I wonder if the outcome of adding
1. A/B/C
2. A/D/E
is correct according to the assumed expected behavior.
All fields seem to be candidates for "final".
Telling the syntax of the key would be nice: absolute file path? Relative file path?

Reply:
Fixed at e36e5a883c9cacde332973b540cfc887e8a73c9f